client-ui/src/app/components/lineage/lineage.component.html

```diff
@@ -18,15 +18,17 @@
     <div class="col-sm-8 no-padding">
       <lineage-graph [embeddedMode]="d.embeddedMode"
                      [layout]="d.layout"
-                     [graph]="d.graph"
+                     [graph]="d.plan.graph"
                      [attributeLineageAndImpactGraph]="d.attributeLinAndImp"
                      [selectedNode]="selectedNodeId"
                      (selectedNodeChange)="onNodeSelected($event)">
       </lineage-graph>
 
       <lineage-highlights-toolbar
               *ngIf="selectedAttribute$ | async as selectedAttributes"
+              [lineageGraphLegends]="lineageGraphLegendsToShow$ | async"
               [attributes]="[selectedAttributes]"
+              [showOldSplineWarning]="isExecPlanFromOldSpline(d.plan.executionPlan)"
```

Comment:
do not use function in the template. You should calculate it in a different way.

Reply:
generally agree, but leaving it as is for now to speed the process up. In this case it's not that important as the function is pure and trivial.